### DISCUSSION TITLE
[Bug] Trace data exported to kafka with null spanId/parentSpanId

### DISCUSSION THREAD

**Discussion by LawrencezZZZ** [apache/skywalking#12528] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

I am using the SkyWalking exporter to export trace data to Kafka. The data obtained from Kafka is split by service rather than by each trace, I can still associate the data using the traceId. The problem is that in the spans of each piece of data, the spanId and parentSpanId are often null, making it difficult for me to link the data together.

### What you expected to happen

All spans should with correct spanId and parentSpanId.

### How to reproduce

Export trace date to kafka

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
https://github.com/apache/skywalking/blob/85b03402e44bd2fdd9499bd20de3ddbf17e41707/oap-server/exporter/src/main/java/org/apache/skywalking/oap/server/exporter/provider/kafka/trace/KafkaTraceExporter.java#L43


We never changed the values of the spans. What you received in proto, is what they are reported.

**Reply by LawrencezZZZ** [reply]
Why do the spans returned by the interface contain correct data when I search for a traceId in SkyWalking UI, but the spans for the same traceId obtained from Kafka lack the spanId and parentSpanId?

**Reply by wu-sheng** [reply]
I don't think there are any difference. Maybe you are using the wrong proto? That is the tricky thing of protobuf. It never complains for field mismatch, it skips those.

**Reply by wu-sheng** [reply]
> Why do the spans returned by the interface contain correct data when I search for a traceId in SkyWalking UI, but the spans for the same traceId obtained from Kafka lack the spanId and parentSpanId?

Again, there isn't lack of xxx, we didn't change that object.
